web/src/app/forms/FormContainer.js

```diff
@@ -1,97 +1,64 @@
-import React from 'react'
-import p from 'prop-types'
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
 import MountWatcher from '../util/MountWatcher'
 
 import { FormContext, FormContainerContext } from './context'
 import { get, set, cloneDeep } from 'lodash'
 
 // FormContainer handles grouping multiple FormFields.
 // It works with the Form component to handle validation.
-export class FormContainer extends React.PureComponent {
-  static propTypes = {
-    value: p.object,
-
-    errors: p.arrayOf(
-      p.shape({
-        message: p.string.isRequired,
-        field: p.string.isRequired,
-        helpLink: p.string,
-      }),
-    ),
-
-    onChange: p.func,
-    disabled: p.bool,
-
-    mapValue: p.func,
-    mapOnChangeValue: p.func,
-
-    // If true, will render optional fields with `(optional)` appended to the label.
-    // In addition, required fields will not be appended with `*`.
-    optionalLabels: p.bool,
-
-    // Enables functionality to remove an incoming value at it's index from
-    // an array field if the new value is falsey.
-    removeFalseyIdxs: p.bool,
-  }
-
-  static defaultProps = {
-    errors: [],
-    value: {},
-    onChange: () => {},
-
-    mapValue: (value) => value,
-    mapOnChangeValue: (value) => value,
-  }
 
-  state = {
-    validationErrors: [],
-  }
+export function FormContainer(props) {
+  const setState = useRef().current
+  const _fields = useRef().current
+  const { value, mapValue, optionalLabels, disabled: containerDisabled } = props
+  const { disabled: formDisabled, addSubmitCheck } = useContext(FormContext)
 
-  _fields = {}
+  useEffect(() => {
+    addSubmitCheck(onSubmit)
+  }, [])
```

Comment:
Make sure we're returnign the value so it gets cleaned up on unmount
```suggestion
  useEffect(() => addSubmitCheck(onSubmit), [])
```

Reply:
Thank you so muck for the feedback! Could you explain to me what the difference between  `  useEffect(() => {
    addSubmitCheck(onSubmit)
  }, [])` and `useEffect(() => addSubmitCheck(onSubmit), [])` in terms of functionality. Thank you!

Reply:
Sure, no problem:

The first one discards the return value
```js
// these are the same

useEffect(() => { addSubmitCheck(onSubmit) }, [])

useEffect(() => {
  const cleanup = addSubmitCheck(onSubmit)

  // returns nothing/undefined
  return
}, [])
```


The second passes the return value through
```js
// these are the same
useEffect(() => addSubmitCheck(onSubmit), [])

useEffect(() => {
  const cleanup = addSubmitCheck(onSubmit)
  return cleanup
}, [])
```

Another way to think of it is `() => foo()` is the same as `() => { return foo() }`